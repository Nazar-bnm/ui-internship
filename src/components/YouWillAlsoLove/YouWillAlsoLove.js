import React from 'react';

import { Button } from '@/shared';
import './YouWillAlsoLove.scss';

const CN = 'you-will-also-love';

const YouWillAlsoLove = (props) => {
  const { userCart, products } = props;

  if (!userCart.length) {
    return (
      <div className={`${CN} content`}>
        <h2 className={`${CN}__title`}>you will also love</h2>
        Buy something please!
      </div>
    );
  }

  if (userCart.length) {
    const selectedGender = userCart[0].data.genders[0];
    const selectedCategory = userCart[0].data.category;
    const selectedSize = userCart[0].data.sizes[0];
    const productsToRender = products.filter((item) => item.category === selectedCategory && item.genders.includes(selectedGender) && item.sizes.includes(selectedSize));
    const itemsToRenderNumber = 3;
    const croppedProducts = productsToRender.slice(0, itemsToRenderNumber);

    const renderProducts = () => (
      croppedProducts.map((item) => {
        const { title, price } = item;
        const imageClaudinaryId = item.images[0].claudinaryId;
        const priceToRender = `$${price}.00`;
        const imageSrc = `${process.env.IMAGE_URL}/${imageClaudinaryId}`;

        return (
          <div className={`${CN}__item`}>
            <img
              alt="product"
              className={`${CN}__image`}
              src={imageSrc}
            />
            <div className={`${CN}__item__description`}>
              <h3 className={`${CN}__item__title`}>{title}</h3>
              <span className={`${CN}__item__price`}>{priceToRender}</span>
              <Button className={`${CN}__quickview-button`}>quickview</Button>
            </div>
          </div>
        );
      })
    );

    return (
      <div className={`${CN} content`}>
        <h2 className={`${CN}__title`}>you will also love</h2>
        {renderProducts()}
      </div>
    );
  }
};

export default YouWillAlsoLove;
