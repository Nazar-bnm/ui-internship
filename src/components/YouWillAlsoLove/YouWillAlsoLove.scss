@import '../../scss/global_styles';

.you-will-also-love {
  @include flex($justify: space-between);
  padding: 20px;
  background-color: $alabaster;
  box-shadow: 0 0 3px $mercury;

  &__title {
    width: 100%;
    margin: 20px;
    font-size: 1rem;
    text-transform: uppercase;
  }

  &__item {
    @include flex($justify: space-between);
    width: 30%;
    margin: 10px;

    &__title {
      font-size: 0.8rem;
      text-transform: uppercase;
    }

    &__description {
      @include flex($direction: column, $justify: space-between);
      width: 50%;
      padding-bottom: 20px;
    }

    &__price {
      margin: 10px 0;
      font-size: 0.8rem;
      font-family: font(primary-light);
    }
  }

  &__image {
    width: 45%;
  }

  &__quickview-button {
    height: 30px;
    border: none;
    box-shadow: 0 0 3px $mercury;
    text-transform: uppercase;

    &:hover {
      transform: scale(1.1);
    }
  }
}