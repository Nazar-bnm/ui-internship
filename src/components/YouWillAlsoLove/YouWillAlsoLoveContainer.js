import { connect } from 'react-redux';

import YouWillAlsoLove from './YouWillAlsoLove';

const mapStateToProps = ({ reducerCart, productsReducer }) => ({
  userCart: reducerCart.userCart,
  products: productsReducer.products
});

export default connect(mapStateToProps, null)(YouWillAlsoLove);
