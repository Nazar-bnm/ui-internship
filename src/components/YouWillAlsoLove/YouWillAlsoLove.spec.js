import React from 'react';
import { shallow } from 'enzyme';

import YouWillAlsoLove from './YouWillAlsoLove';

describe('<ProductImage />', () => {
  test('should match snapshot', () => {
    const products = [{
      chosenQuantity: 1,
      color: "black",
      colors: ["black"],
      data: {
        brandId: "5daf1650a3f2fe001790b09b",
        category: "shirts",
        colors: ["black"],
        createdAt: "2019-10-24T10:15:18.279Z",
        description: "Adopt breathable fabric, soft, lightweight, comfortable to wear. Simple stripe design will make you fashionable and beautiful. It can easily match with denim shorts, close-fitting pants, etc. Suitable for daily wear, perfectly displaying your temperament.",
        genders: ["woman"],
        price: 200,
        quantity: 10,
        seasons: ["spring, summer, autumn"],
        sellCount: 0,
        sizes: ["S", "M"],
        title: "Black shirt with white stripes",
        _id: "5db179b634a29600172f6e8a"
      },
      description: "Adopt breathable fabric, soft, lightweight, comfortable to wear. Simple stripe design will make you fashionable and beautiful. It can easily match with denim shorts, close-fitting pants, etc. Suitable for daily wear, perfectly displaying your temperament.",
      id: "5db179b634a29600172f6e8a",
      price: 200,
      quantity: 10,
      size: "S",
      sizes: ["S", "M"],
      title: "Black shirt with white stripes",
    }];
    const userCart = [{
      chosenQuantity: 1,
      color: "black",
      colors: ["black"],
      data: {
        brandId: "5daf1650a3f2fe001790b09b",
        category: "shirts",
        colors: ["black"],
        createdAt: "2019-10-24T10:15:18.279Z",
        description: "Adopt breathable fabric, soft, lightweight, comfortable to wear. Simple stripe design will make you fashionable and beautiful. It can easily match with denim shorts, close-fitting pants, etc. Suitable for daily wear, perfectly displaying your temperament.",
        genders: ["woman"],
        price: 200,
        quantity: 10,
        seasons: ["spring, summer, autumn"],
        sellCount: 0,
        sizes: ["S", "M"],
        title: "Black shirt with white stripes",
        _id: "5db179b634a29600172f6e8a"
      },
      description: "Adopt breathable fabric, soft, lightweight, comfortable to wear. Simple stripe design will make you fashionable and beautiful. It can easily match with denim shorts, close-fitting pants, etc. Suitable for daily wear, perfectly displaying your temperament.",
      id: "5db179b634a29600172f6e8a",
      price: 200,
      quantity: 10,
      size: "S",
      sizes: ["S", "M"],
      title: "Black shirt with white stripes",
    }];
    const wrapper = shallow(<YouWillAlsoLove userCart={userCart} products={products} />);

    expect(wrapper).toMatchSnapshot();
  });
});
