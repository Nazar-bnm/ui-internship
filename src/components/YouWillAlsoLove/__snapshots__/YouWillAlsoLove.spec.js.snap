// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductImage /> should match snapshot 1`] = `
<div
  className="you-will-also-love content"
>
  <h2
    className="you-will-also-love__title"
  >
    you will also love
  </h2>
</div>
`;
